### PR TITLE
#134676705 notifications

### PR DESCRIPTION
**Pull request by emekanwankwo**
#### What does this PR do?
Send SMS notification to buyers and vendors for products ordered and shipped.
Display notifications on the user dashboard.

#### How should this be manually tested?
Register a user and create an order

#### What are the relevant pivotal tracker stories?
#134676705 notifications

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -3,3 +3,5 @@ FROM reactioncommerce/base:latest
 # Default environment variables
 ENV ROOT_URL "http://localhost"
 ENV MONGO_URL "mongodb://127.0.0.1:27017/reaction"
+ENV JUSIBE_PUBLIC_KEY "0c20d915e92b2718e50be245d350db64"
+ENV JUSIBE_TOKEN "112e8722137c75496b93b7ce80713a68"
```
Comment (femioladeji) [inline Dockerfile]: Is this a safe thing to do?

File: circle.yml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 machine:
   node:
-    version: 4.6.0
+    version: 6.7.0
```
Comment (femioladeji) [inline circle.yml]: Is this advisable? Cos the circleCI build is failing currently and it's probably cos of this?

File: docker-compose.yml [hunk 3]
```diff
@@ -10,6 +10,8 @@ reaction:
   environment:
     ROOT_URL: "http://localhost"
     MONGO_URL: "mongodb://mongo:27017/reaction"
+    JUSIBE_PUBLIC_KEY: "0c20d915e92b2718e50be245d350db64"
+    JUSIBE_TOKEN: "112e8722137c75496b93b7ce80713a68"
```
Comment (femioladeji) [inline docker-compose.yml]: Same as earlier comment about security

File: imports/plugins/core/ui-navbar/client/components/navbar/navbar.js [hunk 4]
```diff
@@ -1,9 +1,82 @@
...
 Template.CoreNavigationBar.onCreated(function () {
   this.state = new ReactiveDict();
+  this.notification = ReactiveVar();
+
+  // Create an auto run to Check for notifications on page load
+  // and set the notification reactive variable.
+  this.autorun(() => {
+    const instance = this;
+    Meteor.call("notifications/getNotifications", Meteor.userId(), (err, res) => {
+      instance.notification.set(!!res);
```
Comment (femioladeji) [inline imports/plugins/core/ui-navbar/client/components/navbar/navbar.js:76]: What's the `!!` for? Is `!!res == res`

File: Dockerfile (hunk 1 above)
Comment (emekanwankwo) [reply Dockerfile]: Thanks for pointing it out, I will find other possible ways to implement it.

File: circle.yml (hunk 2 above)
Comment (emekanwankwo) [reply circle.yml]: I tried updating the node version as a result of the circleci build failing, obviously, it seems that is not the issue.

File: imports/plugins/core/ui-navbar/client/components/navbar/navbar.js (hunk 4 above)
Comment (emekanwankwo) [reply imports/plugins/core/ui-navbar/client/components/navbar/navbar.js:76]: It sets the object instance to boolean.

File: imports/plugins/core/ui-navbar/client/components/navbar/navbar.js [hunk 5]
```diff
@@ -1,9 +1,82 @@
...
+Template.notificationDropdown.helpers({
+  NotificationIcon() {
+  // Check if the user has pending notifications
+  // and set the appropriate Icon
+    return (Template.instance().notification.get() > 0)
+    ? "fa fa-bell"
+    : "fa fa-bell-o";
+  },
+  checkNotification() {
+    return (Template.instance().notification.get() > 0)
```
Comment (azemoh) [inline imports/plugins/core/ui-navbar/client/components/navbar/navbar.js]: This is a Boolean expression which will resolve to `True` or `False`. No need for additional logic.

File: imports/plugins/core/ui-navbar/client/components/navbar/navbar.js [hunk 6]
```diff
@@ -1,9 +1,82 @@
...
+});
+Template.notificationItem.helpers({
+  showNotification() {
+    // Change the display state of the notification to show the latest notification when clicked
+    return Template.instance().notification.get();
+  }
+});
 Template.CoreNavigationBar.onCreated(function () {
   this.state = new ReactiveDict();
+  this.notification = ReactiveVar();
```
Comment (azemoh) [inline imports/plugins/core/ui-navbar/client/components/navbar/navbar.js:69]: If this is an Array of notifications, I think the variable name should reflect that. i.e `notifications`

File: imports/plugins/core/ui-navbar/client/components/navbar/navbar.js [hunk 7]
```diff
@@ -23,6 +96,8 @@ Template.CoreNavigationBar.events({
     }, $("body").get(0));
     $("body").css("overflow", "hidden");
     $("#search-input").focus();
+  },
+  "click .notifications": function (events, template) {
```
Comment (azemoh) [inline imports/plugins/core/ui-navbar/client/components/navbar/navbar.js]: Event handler with empty logic?

File: imports/plugins/core/ui-navbar/client/components/navbar/navbar.js (hunk 6 above)
Comment (emekanwankwo) [reply imports/plugins/core/ui-navbar/client/components/navbar/navbar.js:69]: Thanks for pointing this out, I have made the changes to other instances where it is an array, however, this particular case is not an array.

File: package.json [hunk 8]
```diff
@@ -35,6 +35,7 @@
     "css-annotation": "^0.6.2",
     "deep-diff": "^0.3.4",
     "dnd-core": "^2.0.2",
+    "dotenv": "^2.0.0",
```
Comment (azemoh) [inline package.json]: Remove unused package.

**Pull request merged**